packages/aws-rfdk/lib/deadline/lib/render-queue.ts

```diff
@@ -465,16 +465,38 @@ export class RenderQueue extends RenderQueueBase implements IGrantable {
    *
    * See: https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html for additonal information.
    *
-   * @param includeResourceTracker Whether or not the Resource tracker admin policy should also be addd (Default: True)
+   * @param includeResourceTracker Whether or not the Resource tracker admin policy should also be added (Default: True)
+   * @param iamFleetRoleArns Allows render queue to use custom spot fleet roles (Default: only aws-ec2-spot-fleet-tagging-role is allowed)
    */
-  public addSEPPolicies( includeResourceTracker: boolean = true): void {
+  public addSEPPolicies(includeResourceTracker: boolean = true, iamFleetRoleArns: string[] | undefined = undefined): void {
     const sepPolicy = ManagedPolicy.fromAwsManagedPolicyName('AWSThinkboxDeadlineSpotEventPluginAdminPolicy');
     this.taskDefinition.taskRole.addManagedPolicy(sepPolicy);
 
     if (includeResourceTracker) {
       const rtPolicy = ManagedPolicy.fromAwsManagedPolicyName('AWSThinkboxDeadlineResourceTrackerAdminPolicy');
       this.taskDefinition.taskRole.addManagedPolicy(rtPolicy);
     }
+
+    this.taskDefinition.taskRole.addToPrincipalPolicy(new PolicyStatement({
+      actions: [
+        'ec2:CreateTags',
+      ],
+      resources: ['arn:aws:ec2:*:*:spot-fleet-request/*'],
+    }));
+
+    if (iamFleetRoleArns) {
+      this.taskDefinition.taskRole.addToPrincipalPolicy(new PolicyStatement({
```

Comment:
`AWSThinkboxDeadlineSpotEventPluginAdminPolicy` has `iam:PassRole` action allowed only for a default rolename `aws-ec2-spot-fleet-tagging-role`, so if we use a role with a different name, then role validation will fail.